### PR TITLE
Adding module to manage replication subnet groups for AWS Database Migration Service

### PR DESCRIPTION
**Pull request by slapula**
##### SUMMARY
This is my first pull request of several to add support for AWS Database Migration Service.  This module manages the replication subnet groups in the spirit of `rds_subnet_group` and `redshift_subnet_group`.  It's a pretty simple module in terms of functionality.  One omission however.  I did not include support for adding and removing tags since AWS' documentation around replication subnet group ARNs is lacking (and I haven't been able to figure out how to properly form them myself) and you need the ARN to use the API calls.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
`dms_subnet_group`

##### ANSIBLE VERSION
```
ansible 2.4.1.0
  config file = None
  configured module search path = [u'/Users/asmith/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/asmith/.pyenv/versions/2.7.13/lib/python2.7/site-packages/ansible
  executable location = /Users/asmith/.pyenv/versions/2.7.13/bin/ansible
  python version = 2.7.13 (default, Sep  8 2017, 15:16:38) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in integration tests due to traceback:

https://app.shippable.com/github/ansible/ansible/runs/66096/67/tests

```
Traceback (most recent call last):
  File "/tmp/ansible_qexercul/ansible_module_ec2_vpc_net.py", line 180, in vpc_exists
    matching_vpcs = vpc.describe_vpcs(Filters=[{'Name': 'tag:Name', 'Values': [name]}, {'Name': 'cidr-block', 'Values': cidr_block}])['Vpcs']
  File "/tmp/ansible_qexercul/ansible_modlib.zip/ansible/module_utils/aws/core.py", line 215, in deciding_wrapper
    return unwrapped(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (AuthFailure) when calling the DescribeVpcs operation: AWS was not able to validate the provided access credentials
```

**Comment by mattclay** [comment]
@s-hertel Is this something we can support testing in CI instead of marking it `unsupported`?

File: test/integration/targets/dms_subnet_group/aliases [hunk 1]
```diff
@@ -0,0 +1,3 @@
+cloud/aws
+unsupported
+unstable
```
Comment (mattclay) [inline test/integration/targets/dms_subnet_group/aliases]: New tests should not be marked `unstable`. Also, the tests should only be marked `unsupported` if we cannot add permissions to our CI system to run the tests or the tests will be too slow to be used in CI.

**Comment by Akasurde** [comment]
Please submit a new module request to https://github.com/ansible-collections/community.aws. Thanks for your contribution.

**Pull request closed without merging**